cmd/kogito/command/remove/factory.go

```diff
@@ -20,12 +20,10 @@ import (
 )
 
 // BuildCommands creates the commands available in this package
-func BuildCommands(ctx *context.CommandContext, rootCommand *cobra.Command) []context.KogitoCommand {
-	removeCmd := newRemoveCommand(ctx, rootCommand)
-	return []context.KogitoCommand{
-		removeCmd,
-		newRemoveInfinispanCommand(ctx, removeCmd.Command()),
-		newRemoveKeycloakCommand(ctx, removeCmd.Command()),
-		newRemoveKafkaCommand(ctx, removeCmd.Command()),
-	}
+func BuildCommands(ctx *context.CommandContext, rootCommand *cobra.Command) {
+	removeCmd := initRemoveCommand(ctx, rootCommand)
```

Comment:
Since we're now modifying an existing command rather than creating a new one (replaced the **new** methods to **init** methods), for me it makes more sense to call these **init** methods like:

```go
rootCommand.initRemoveCommand(ctx)
   .initRemoveXXXCommand(ctx)
   ...
```

Also, I think we could delegate these initializers using the decorator pattern or similar. So, we just call a list of initializers and each initializer will be for either Infinispan, Keycloak, Kafka or any other in the future.

Reply:
Hi! Thanks for the feedback! Please fell free to create a JIRA for 0.10.0 version about this refactoring.

Reply:
Raised: https://issues.redhat.com/browse/KOGITO-1720. Thanks!